src/Chapter3.hs

```diff
@@ -971,6 +1146,39 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+days = ["Sun", "Mon","Tue" ,"Wed", "Thu", "Fri", "Sat"]
+
+data WeekDay = WeekDay
+  {
+    day :: String
+  }
+
+isWeekend :: String -> Bool
+isWeekend "Sun" = True
+isWeekend "Sat" = True
+isWeekend _ = False
+
+-- Brute-forcing this
+nextDay :: String -> String
+nextDay "Sun" = "Mon"
+nextDay "Mon" = "Tue"
+nextDay "Tue" = "Wed"
+nextDay "Wed" = "Thu"
+nextDay "Thu" = "Fri"
+nextDay "Fri" = "Sat"
+nextDay "Sat" = "Sun"
```

Comment:
If you change the type to the sum type, you would be able to use `Bounded` typeclass instance to implement this function easily and safely

Reply:
@vrom911 Can you please point me to code that does this ? I actually got stuck here.

Reply:
Sure, take a look at [`succ`](https://hackage.haskell.org/package/base/docs/Prelude.html#v:succ) and [`maxBound`](https://hackage.haskell.org/package/base/docs/Prelude.html#v:maxBound) + [`minBound`](https://hackage.haskell.org/package/base/docs/Prelude.html#v:minBound).

`succ` will give you the next constructor, but you should be careful about the last constructor. For that, you could check on the `maxBound` and return `minBound` in that case 🙂